### PR TITLE
Refactored Config

### PR DESCRIPTION
**Pull request by SidRoberts**
Hello!

* Type: code quality

**In raising this pull request, I confirm the following:**

- [x] I have read and understood the [Contributing Guidelines](https://github.com/phalcon/cphalcon/blob/master/CONTRIBUTING.md)
- [x] I have checked that another pull request for this purpose does not exist
- [-] I wrote some tests for this PR
- [-] I updated the CHANGELOG

Small description of change:

Thanks

### CODE REVIEW

**Review by Jeckerson — DISMISSED** [review]

**Review by niden — APPROVED** [review]

**Comment by niden** [comment]
@SidRoberts I would like to refactor this component to start using the `Collection` class in say 4.1. Less duplication of code.

Thanks for this change

File: phalcon/Config.zep [hunk 1]
```diff
@@ -215,8 +215,7 @@ class Config implements \ArrayAccess, \Countable
     {
         let index = strval(index);
 
-        //unset(this->{index});
-        let this->{index} = null;
+        unset this->{index};
```
Comment (ekmst) [inline phalcon/Config.zep]: @niden @SidRoberts 
Compile error.
 [ERROR] Cannot use expression type: property-dynamic-access in "unset" in cphalcon/phalcon/Config.zep on line 218

File: phalcon/Config/Factory.zep [hunk 2]
```diff
@@ -74,27 +74,27 @@ class Factory extends BaseFactory
...
-                    return new {className}(filePath, callbacks);
-                }
-            }
+        let className = namespaceClass . "\\" . camelize(adapter);
+        if !strpos(filePath, ".") {
+            let filePath = filePath . "." . lcfirst(adapter);
+        }
 
-            return new {className}(filePath);
+        if className === \Phalcon\Config\Adapter\Ini::class {
```
Comment (ekmst) [inline phalcon/Config/Factory.zep]: @niden @SidRoberts 
Compile error.
Syntax error in cphalcon/phalcon/Config/Factory.zep on line 88.
I think, this is must be "Phalcon\\Config\\Adapter\\Ini"

File: phalcon/Config/Factory.zep [hunk 3]
```diff
@@ -74,27 +74,27 @@ class Factory extends BaseFactory
...
+            if fetch mode, config["mode"] {
+                return new {className}(filePath, mode);
+            }
+        } elseif className === \Phalcon\Config\Adapter\Yaml::class {
```
Comment (ekmst) [inline phalcon/Config/Factory.zep]: @niden @SidRoberts 
Compile error.
Syntax error in cphalcon/phalcon/Config/Factory.zep on line 92.
I think, this is must be "Phalcon\\Config\\Adapter\\Yaml"

File: phalcon/Config/Factory.zep (hunk 2 above)
Comment (SidRoberts) [reply phalcon/Config/Factory.zep]: I've made https://github.com/phalcon/cphalcon/pull/13951 to fix it. I'm still waiting on AppVeyor but it looks to be passing.

File: phalcon/Config/Factory.zep (hunk 3 above)
Comment (SidRoberts) [reply phalcon/Config/Factory.zep]: I've made https://github.com/phalcon/cphalcon/pull/13951 to fix it. I'm still waiting on AppVeyor but it looks to be passing.

**Comment by niden** [comment]
@ekmst @SidRoberts sorted it out. Thanks for keeping an eye out.

**Pull request merged**